CONTRIBUTING.md

```diff
@@ -15,42 +14,103 @@
 
 ## How to write tests
 
 ### Differentiate between NodeJS runtimes
 
 When developing in Cypress, you need to understand two different JavaScript runtimes
 
-* The one Cypress bundles into the browser under test
-* System wide JavaScript
+- The one Cypress bundles into the browser under test
+- System wide JavaScript
+
+Since using TestContainers to launch containers shouldn't be bundled into browser under test, starting the WildFly container is being done using Cypress arbitrary Node code running handle - [`cy.task`](https://docs.cypress.io/api/commands/task), use
 
-Since using TestContainers to launch containers shouldn't be bundled into browser under test, starting the WildFly container is being done using Cypress arbitrary Node code running handle - [`cy.task`](https://docs.cypress.io/api/commands/task), use 
 ```
 cy.startWildFlyContainer
 ```
+
 method, which is a wrapper for
+
 ```
 cy.task("start:wildfly:container")
 ```
-to start the container. This method/task also automatically sets the deployed HAL container endpoint in the `allowed-origins` attribute. Both of the approaches would return the management endpoint of the deployed WildFly instance.
+
+to start the container. Note that despite starting the web server being marked as an anti-pattern, starting the WildFly server clean has its benefits in terms of a clean environment per test and also it introduces means for test parallelization (we know what we're doing :slightly_smiling_face:). This method/task also automatically sets the deployed HAL container endpoint in the `allowed-origins` attribute. Both of the approaches would return the management endpoint of the deployed WildFly instance.
 
 ### Communicating with WildFly management interface
 
 Communication with the WildFly management interface is also being done using Cypress task, `execute:cli` to be precise. A better example of running `/system-property=to-add:add(value=myVal)` in JBoss CLI in Berg using `cy.task`
-would be 
+would be
+
 ```
 cy.task("execute:cli", {
       managementApi: `${managementEndpoint}/management`,
       operation: "add",
       address: ["system-property", "to-add"],
       value: "myVal"
     });
-```
\ No newline at end of file
+```
+
+### Test template
+
+To start developing tests, you can copy paste following example into your `packages/testsuite/cypress/e2e/test-to-be-added.cy.ts`
+
+```typescript
+// describe method is an equivalent of a TestClass
+describe("TESTS: Path => To => Tested => Resource", () => {
+  const configurationFormId = "resource-form";
+
+  // This is and before method are very important
+  let managementEndpoint: string;
+
+  // Here we start our WildFly container prior to test execution
+  before(() => {
+    // Obtains WildFly management endpoint, that will be used
+    // for backend data retrieval and validation
+    // cy.startWildflyContainer also starts the WildFly container under
```

Comment:
> starts the WildFly container under the name of the spec/test file

I don't understand the impact. Can you elaborate/clarify please?

Reply:
@OndrejKotek I've elaborated, can this be closed? 🙂